clients/src/test/java/org/apache/kafka/common/SerializeCompatibilityTopicPartition.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common;
+
+import org.apache.kafka.common.utils.Serializer;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This test ensures TopicPartition class is serializable and is serialization compatible.
+ * Note: this ensures only serialization compatibility backwards but would not be able to ensure the reverse i.e.
+ * deserialize a object of new version of the class into a Old version class will not be ensured.
+ */
+public class SerializeCompatibilityTopicPartition extends Serializer {
+
+    private String topicName = "mytopic";
+    private int    partNum   = 5;
+    private String fileName  = "serializedData/topicPartitionSerializedfile";
```

Comment:
Could you let me know what exactly are you referring to here? Not sure I get what u point at when said block formatting.

Do you refer to multiple spaces between datatype declaration and the variable name? (example space between int and partNum)